stdlib/_ssl.pyi

```diff
@@ -0,0 +1,297 @@
+
+if sys.version_info < (3, 10):
+    def RAND_egd(path: str) -> None: ...
+
+def RAND_status() -> bool: ...
+def get_default_verify_paths() -> tuple[str, str, str, str]: ...
+
+if sys.platform == "win32":
+    _EnumRetType: TypeAlias = list[tuple[bytes, str, set[str] | bool]]
+    def enum_certificates(store_name: str) -> _EnumRetType: ...
+    def enum_crls(store_name: str) -> _EnumRetType: ...
+
+def txt2obj(txt: str, name: bool = False) -> tuple[int, str, str, str]: ...
+def nid2obj(__nid: int) -> tuple[int, str, str, str]: ...
+
+class _SSLContext:
+    check_hostname: bool
+    if sys.version_info >= (3, 8):
+        keylog_filename: str | None
+    maximum_version: int
+    minimum_version: int
+    num_tickets: int
+    options: int
+    if sys.version_info >= (3, 8):
+        post_handshake_auth: bool
+    protocol: int
+    if sys.version_info >= (3, 10):
+        security_level: int
+    sni_callback: Callable[[SSLObject, str, SSLContext], None | int] | None
+    verify_flags: int
+    verify_mode: int
+    def __new__(cls, __protocol: int) -> Self: ...
+    def cert_store_stats(self) -> dict[str, int]: ...
+    @overload
+    def get_ca_certs(self, binary_form: Literal[False] = False) -> list[_PeerCertRetDictType]: ...
+    @overload
+    def get_ca_certs(self, binary_form: Literal[True]) -> list[bytes]: ...
+    @overload
+    def get_ca_certs(self, binary_form: bool = False) -> Any: ...
+    def get_ciphers(self) -> list[_Cipher]: ...
+    def load_cert_chain(
+        self, certfile: StrOrBytesPath, keyfile: StrOrBytesPath | None = None, password: _PasswordType | None = None
+    ) -> None: ...
+    def load_dh_params(self, __path: str) -> None: ...
+    def load_verify_locations(
+        self,
+        cafile: StrOrBytesPath | None = None,
+        capath: StrOrBytesPath | None = None,
+        cadata: str | ReadableBuffer | None = None,
+    ) -> None: ...
+    def session_stats(self) -> dict[str, int]: ...
+    def set_ciphers(self, __cipherlist: str) -> None: ...
+    def set_default_verify_paths(self) -> None: ...
+    def set_ecdh_curve(self, __name: str) -> None: ...
+
+@final
+class MemoryBIO:
+    eof: bool
+    pending: int
+    # stubtest thinks this is (self, *args, *kwargs)
+    # and doesn't respect the alowlist entry for some reason
+    # def __init__(self) -> None: ...
+    def read(self, __size: int = -1) -> bytes: ...
+    def write(self, __b: ReadableBuffer) -> int: ...
+    def write_eof(self) -> None: ...
+
+@final
+class SSLSession:
+    has_ticket: bool
+    id: bytes
+    ticket_lifetime_hint: int
+    time: int
+    timeout: int
+    def __eq__(self, other: object) -> bool: ...
+    def __ge__(self, other: object) -> bool: ...
+    def __gt__(self, other: object) -> bool: ...
+    def __le__(self, other: object) -> bool: ...
+    def __lt__(self, other: object) -> bool: ...
+    def __ne__(self, other: object) -> bool: ...
+
+# _ssl.Certificate is weird: it can't be instantiated or subclassed.
```

Comment:
@AlexWaygood I removed _ssl._SSLObject which underlies ssl.SSLObject but isn't a formal base class for it. Doing so leaves _ssl.Certificate a little stranded.

It don't have an underscore prefix, so stubcheck complains if it's absent, but the only possible way to create an instance is via a method of _ssl._SSLSocket. Part of the reason I fleshed out the stubs for _SSLSocket in the first place was that I was trying to figure how to generate an instance of Certificate in order to observe its behavior.

I waffled on leaving the _SSLSocket stubs in, but I figured this comment would have the same effect without (as much) maintenance burden. On the other hand, that's probably outside the goals of typeshed, so I understand if it'd be better to remove the comment and just let Certificate be an oddity with no clear explanation.